src/Orleans/Async/Cancellation/GrainCancellationTokenSource.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.Async
+{
+    /// <summary>
+    /// Distributed version of the CancellationTokenSource
+    /// </summary>
+    public class GrainCancellationTokenSource : IDisposable
```

Comment:
Make sure we never send that around. Implement special serialization methdo to it that will throw.